Improve error message for admin yml validation

yakky opened the pull request:
If the yml validation fails, we report the missing field and the dump of the yml part that fails validation. This will help admins to provide information to publisher maintainers

As a bonus, it adds a command to parse yml files for local debug

yakky committed:
Improve error message for admin yml validation

If the yml validation fails, we report the missing field and the dump of the yml part that fails validation. This will help admins to provide information to publisher maintainers

francescozaia commented:
Se lo ritenete sensato, questa PR si può riciclare per https://trello.com/c/FOyE0dIa/93-alcuni-fix-sui-file-yml-di-configurazione.

yakky commented:
@francescozaia sì, l'idea era quella

yakky committed:
Update yml validation logic

This brings the accepted yml structure with the latest revision of configuration specs

yakky commented:
Ho preferito ribasare questa sulla #63 per poter fare funzionare eventuali test sul server di staging, quindi ci ritroviamo il commit 27767acba09d18dfbef7f45201ea471fe84e06ad
appena fatto il merge dell'altra chiudo e riapro questa

xrmx commented on readthedocs/docsitalia/models.py:
```diff
@@ -53,16 +53,14 @@ def validate_projects_metadata(org, settings):
+            for index, document in enumerate(project['documents']):
+                project['documents'][index] = {
+                    'repository':  document,
+                    'repo_url': '{}/{}'.format(org.url, document)
```
the comment was handy though

yakky replied:
Done

xrmx commented on readthedocs/docsitalia/models.py:
```diff
@@ -37,7 +37,7 @@ def validate_publisher_metadata(org, settings): # noqa
                 raise ValueError('Missing required field "%s" in %s' % (field, publisher))
     except (KeyError, TypeError):
-        raise ValueError('General error in parsing %s' % data)
+        raise ValueError('General error in parsing publisher metadata %s' % data)
```
Note to self, we may want to return the validate_*_metadata exception messages in docsitalia.github.parse_metadata instead
